### ISSUE TITLE
Suggestion: Charge state-dependent notifications

### ISSUE THREAD

**Issue by Lord-Memester** [medhachaitanya/Pure-Battery-Analytics#19]
I would like more options for when the program sends notifications.

For example, I have a notification set for when the battery is at 30%. 
- When my laptop is discharging and gets to 30%, sending me a notification is really useful, and lets me know to charge it. 
- If my laptop is at 27% and while charging it reaches 30%, I don't feel the need to know, since I plan to just keep charging it past that point.

So a setting of being notified `while charging`, `while discharging`, or `both` for each of the individual user-specified notification-associated percentages would be greatly appreciated.

**Comment by medhachaitanya** [comment]
I completely agree. I'm currently working on making it simpler. You can now add any desired percentage, using a minus symbol (-) to indicate discharging.

For example:
- **Example 1:** `-30, -50, 80, 85` will show notifications at 30% and 50% when discharging, and at 80% and 85% when charging.
- **Example 2:** `-30, 90` will show notifications at 30% when discharging and at 90% when charging.
- **Example 3:** `-30, 30, 90` will show notifications at 30% when discharging and at 30%, 90% when charging.

In summary, using the minus symbol (-) specifies that the notification should be shown when the battery is discharging.

**Comment by Lord-Memester** [comment]
Excellent! An additional suggestion to consider:

- `-` is used as you suggested.
- `+` prepended to a value will notify only when charging.
- if nothing is prepended to the value, notify on both charging and discharging
- _if `-+` or `+-` is prepended, either disallow this in the program and pop up a warning (like when you try to name a file with invalid characters in windows) or have it notify on both charging and discharging_

I would personally find this preferable, as it would do away with the need for double entry of values and would visually make more sense. It would also be easy to include a short guide for in the window, like below:

---
Default behavior is to notify upon being charged to and discharged to any provided value. 

**For more precise control, prepend:**
- `-` to notify when discharged to the value
- `+` to notify when charged to the value

**Comment by medhachaitanya** [comment]
I like it. Thanks for sharing. I released a build for the aforementioned feature already. I will squeeze some time to work on the new suggestion soon.

**Comment by Lord-Memester** [comment]
> I like it. Thanks for sharing. I released a build for the aforementioned feature already. I will squeeze some time to work on the new suggestion soon.

I must say, I cannot find this build. I assume this has been made part of the Pure Battery Percentage Add-On, but this repository hasn't been updated (had any commits) since last year, and I can find no new release on the Microsoft Store when I check for updates.

**Comment by medhachaitanya** [comment]
It's still under processing. You will get a Microsoft store update sometime in the next couple of hours.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Lord-Memester** [comment]
> It's still under processing. You will get a Microsoft store update sometime in the next couple of hours.

I've gotten the update now! Found an issue though:

A space between the comma and the number registers as an invalid entry, which is contrary to the example of (`-30, 90`) described in the window.
![image](https://github.com/medhachaitanya/Pure-Battery-Analytics/assets/71044282/ca8b5c51-8658-4d92-bcd2-9ffa75954b63) (`-30, 50`)
![image](https://github.com/medhachaitanya/Pure-Battery-Analytics/assets/71044282/f6915f91-ee06-490e-b8a5-d31766ad2cd6) (`-30, -50`)


If we remove the space, it registers as valid.
![image](https://github.com/medhachaitanya/Pure-Battery-Analytics/assets/71044282/b8e6f17d-1842-453d-aa03-2cd3fdb55824) (`-30,50`)
![image](https://github.com/medhachaitanya/Pure-Battery-Analytics/assets/71044282/e219c1e5-8242-4dc2-92c9-c375ca4ba455) (`-30,-50`)

The entry with a space between seems to be the intended method, so this needs to be fixed.
